### ISSUE TITLE
Unable to parse generics on 1.3.0

### ISSUE THREAD

**Issue by kuuuurt** [ktorio/ktor#1579]
**Ktor Version and Engine Used (client or server and name)**
Ktor Version: 1.3.0

**Describe the bug**
My generic classes can't be parsed on 1.3.0

**To Reproduce**
I have this `Response` class

```kotlin
@Serializable
class Response<T> {
    val status: Int,
    val message: String,
    val data: T
}
```

for this response:

```
{
    status: 200,
    message: "Success",
    data: { ... }
}
```
It errors with a message that the only generic classes for now are standard collections.

This works on 1.2.6

**Expected behavior**
It should be able to parse the generic `Response` class

**Comment by denmusic1992** [comment]
I have the same problem with generics, I have the Identical response:
```kotlin
@Serializable
data class CommonResponse <T> (
    @SerialName("Status")
    val status: String,
    @SerialName("Code")
    val code: Int,
    @SerialName("ErrorList")
    val errorList: List<String>?,
    @SerialName("MessageList")
    val messageList: List<String>?,
    @SerialName("Data")
    val data: T?
)
```
and I'm using a **_KotlinxSerializer_** for client:
```kotlin
install(JsonFeature) {
            //serializer =
            serializer = KotlinxSerializer(Json.nonstrict).apply {
                // Перечисляем, какие именно варианты у нас могут быть с generic data параметр
                //register<Device>()
                register(CommonResponse.serializer(User.serializer()))
                register(CommonResponse.serializer(Device.serializer()))

            }
        }
```
_register_ is now Deprecated in 1.3.0. Also it registers only the last KSerializer. @kuuuurt could you please share the code where you register your Response please?

**Comment by kuuuurt** [comment]
I do the exact thing you did. I reverted back to 1.2.6 and kept a separate branch for 1.3.0 for now

**Comment by denmusic1992** [comment]
@kuuuurt did you manage to register multiple serializers for your Response _data_ in 1.2.6? In my case, it only registers the last class....

**Comment by kuuuurt** [comment]
@denmusic1992 yeah, it doesn't work for me too. I create an `HttpClient` everytime and do the registration per call. I just created helper methods so it's easier for me.

**Comment by cy6erGn0m** [comment]
Any example of how do you receive it? What error messages?

**Comment by cy6erGn0m** [comment]
Registration is deprecated and does nothing in 1.3.0

**Comment by kuuuurt** [comment]
Here's the error I'm receiving. 

<img width="347" alt="Screen Shot 2020-01-22 at 11 16 09 AM" src="https://user-images.githubusercontent.com/18444033/72872952-860a5900-3d29-11ea-89b5-3e05ee5f2f7d.png">

**Comment by e5l** [comment]
Hi @kuuuurt, thanks for the report. As I see this is the current limitation of kotlinx.serialization. It'll be fixed in the next serialization release.

**Comment by kuuuurt** [comment]
Thanks @e5l! Will be using 1.2.6 in the meantime

**Comment by kuuuurt** [comment]
Hello, @e5l! Just wondering, is there a link to an issue on serialization referring to this bug?

**Comment by e5l** [comment]
Nope. Fixed in `1.3.2`
